pkg/handlers/generic/lifecycle/config/cm.go

```diff
@@ -89,7 +110,10 @@ func (h *HelmChartGetter) For(
 	if !ok {
 		return nil, fmt.Errorf("did not find key %s in %v", name, cm.Data)
 	}
-	var settings HelmChart
-	err = yaml.Unmarshal([]byte(d), &settings)
-	return &settings, err
+	var chart HelmChart
+	err = yaml.Unmarshal([]byte(d), &chart)
```

Comment:
The [logic](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/497/files#diff-cf72970549260134569fee23f53ae5f544f32b05f043a5da47c4dcd3a35de86eR94) that reads the helmChartRepo can be moved here. If the configmap retrival errors out or the component config does not exists, we can return before reading the helmChartRepo variable.